src/Compute/Compute.Test/ScenarioTests/GalleryTests.ps1

```diff
@@ -893,3 +893,68 @@ function Test-GalleryVersionWithSourceImageVMId
         Clean-ResourceGroup $rgname;
     }
 }
+
+<#
+.SYNOPSIS
+Tests New-AzGalleryImageDefinition to default to HyperVGen V2 and TL
+#>
+function Test-GalleryImageDefinitionDefaults
+{
+    # Setup
+    $rgname = Get-ComputeTestResourceName;
+    $loc = Get-ComputeVMLocation;
+
+    try
+    {
+    
+        $location = $loc;
+        New-AzResourceGroup -Name $rgname -Location $loc -Force;
+
+        # Gallery variables
+        $resourceGroup = $rgname
+        $galleryName = 'gl' + $rgname
+        $definitionName = 'def' + $rgname
+        $skuDetails = @{
+            Publisher = 'test'
+            Offer = 'test'
+            Sku = 'test'
+        }
+        $osType = 'Windows'
+        $osState = 'Specialized'
+        $storageAccountSku = 'Standard_LRS'
+        
+        # Setup Image Gallery
+        New-AzGallery -ResourceGroupName $rgname -Name $galleryName -location $location
+
+        # Setup Image Definition
+        $paramNewAzImageDef = @{
+            ResourceGroupName = $rgname
+            GalleryName       = $galleryName
+            Name              = $definitionName
+            Publisher         = $skuDetails.Publisher
+            Offer             = $skuDetails.Offer
+            Sku               = $skuDetails.Sku
+            Location          = $location
+            OSState           = $osState
+            OsType            = $osType
+            ErrorAction       = 'Stop'
+        }
+        
+        New-AzGalleryImageDefinition @paramNewAzImageDef;
+
+        $definition = Get-AzGalleryImageDefinition -ResourceGroupName $rgname -GalleryName $galleryName -Name $definitionName;
```

Comment:
Since not all galleries or VMs support TL, is there a way for users to turn off this TL defaulting? Like passing in a standard value? Can you test that?

Reply:
added a test case where SecurityType is passed in as ConfidentialVM, and verify  "TrustedLaunchSupported" securityType was not defaulted

Reply:
No, but is there a way to not have a SecurityType? I imagine not all setups are compatible with ConfidentialVM either,

Reply:
this specific case does not provide SecurityType, we check for defaulting to Trusted Launch

Reply:
case below, SecurityType is explicitly provided "ConfidentialVM" and we verify that TL default did not take place